Caboodle/Battery/Battery.ios.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Foundation;
+using UIKit;
+
+namespace Microsoft.Caboodle
+{
+    public static partial class Battery
+    {
+        public static double ChargeLevel
+        {
+            get
+            {
+                var batteryMonitoringEnabled = UIDevice.CurrentDevice.BatteryMonitoringEnabled;
+                UIDevice.CurrentDevice.BatteryMonitoringEnabled = true;
+                try
+                {
+                    return UIDevice.CurrentDevice.BatteryLevel;
+                }
+                finally
+                {
+                    UIDevice.CurrentDevice.BatteryMonitoringEnabled = batteryMonitoringEnabled;
+                }
+            }
+        }
+
+        public static BatteryState State
+        {
+            get
+            {
+                var batteryMonitoringEnabled = UIDevice.CurrentDevice.BatteryMonitoringEnabled;
+                UIDevice.CurrentDevice.BatteryMonitoringEnabled = true;
+                try
+                {
+                    switch (UIDevice.CurrentDevice.BatteryState)
+                    {
+                        case UIDeviceBatteryState.Charging:
+                            return BatteryState.Charging;
+                        case UIDeviceBatteryState.Full:
+                            return BatteryState.Full;
+                        case UIDeviceBatteryState.Unplugged:
+                            return BatteryState.Discharging;
+                        default:
+                            return BatteryState.Unknown;
```

Comment:
We could have a check here to see if `ChargeLevel == 100`, then return full.

Reply:
I think iOS returns Full if that is the case...